src/modules/api/graphql/resolvers.ts

```diff
@@ -39,6 +74,25 @@ export const DebugResolvers = {
 }
 
 export const Resolvers = {
+  contractStateDiff: async (_, args) => {
+    const inputTxMatchingOutputTx = await appContainer.self.newService.chainBridge.contractOutputDb.findOne(
+      { inputs: { $elemMatch: { $eq: args.id } } }
+    );
+
+    // ALP: revisit when deterministic mechanism for arranging order of tx in a block is found, currently not save when multiple executions in block happen
+    const previousContractOutputTx = await appContainer.self.newService.chainBridge.contractOutputDb.findOne(
+      { anchored_height: { $lt: inputTxMatchingOutputTx.anchored_height } },
```

Comment:
Might be able to do the same, but use `_id` instead. I think that is guaranteed to be ordered by the insert order in the Mongo collection